File: codechef/permut2.cpp
#include <iostream>
using namespace std;

const int MAX = 100000;
int arr[MAX];

int main()
{
  int n;
  while(1)
  {
    cin>>n;
    bool ambiguous = true;
    
    if(n==0)
      break;
      
     for(int i=0; i<n; i++)
     {
        cin>>arr[i];
     }
     
     for(int i=0; i<n; i++)
     {
        if(arr[i] > n || arr[arr[i]-1] != (i+1))
        {
            ambiguous = false;
            break;
        }
     }
     
     cout<<(ambiguous ? "ambiguous" : "not ambiguous")<<endl; 
  }

  return 0;
}

File: spoj/permut2.py
#
# Problem: http://www.spoj.com/problems/PERMUT2/
# Category: Simple coding
#
while True:
    n = int(raw_input())
    if n == 0:
        break
    arr = map(int, raw_input().split())
    ans = 'ambiguous'
    for i in xrange(1, n + 1):
        p = arr[i - 1]
        if arr[p - 1] != i:            
            ans = 'not ambiguous'
            break
    print ans    